checker/src/org/checkerframework/checker/index/upperbound/UpperBoundTransfer.java

```diff
@@ -46,8 +47,20 @@ public UpperBoundTransfer(CFAnalysis analysis) {
             AssignmentNode node, TransferInput<CFValue, CFStore> in) {
         TransferResult<CFValue, CFStore> result = super.visitAssignment(node, in);
 
-        if (node.getExpression() instanceof ArrayCreationNode) {
-            ArrayCreationNode acNode = (ArrayCreationNode) node.getExpression();
+        Node expNode = node.getExpression();
+
+        ArrayCreationNode acNode =
+                expNode instanceof ArrayCreationNode ? (ArrayCreationNode) expNode : null;
+
+        if (expNode instanceof TypeCastNode) {
```

Comment:
Actually, it can't call `TreeUtils.skipParens`, because it's a `Node` and we want it to stay at that level. Further, I don't see a ParenNode or anything similar; does such a thing even exist at the Node level?